napari/_qt/utils.py

```diff
@@ -469,3 +471,14 @@ def qt_might_be_rich_text(text) -> bool:
         return _Qt.mightBeRichText(text)
     except Exception:
         return bool(RICH_TEXT_PATTERN.search(text))
+
+
+def check_if_in_main_thread():
+    """Check if we are in the main thread.
+
+    Returns
+    -------
+    thread_flag : bool
+        True if we are in the main thread, False otherwise.
+    """
+    return QCoreApplication.instance().thread() == QThread.currentThread()
```

Comment:
I do not think earlier about main_thread. 

In general it is not good solution (as main thread in python is thread in which start interpreter, and Qt main threat is thread where Application object is created. 

But as it is for debugging purpose, then, with proper explanation it could be done in this way. I need to rethink this.

Reply:
I do not expect but this is possible. But I'm not sure if we need to support it.

Reply:
I second Andy's sentiments here. Mostly, it'd be nice to new Qt dependancies in otherwise non-graphical code.

Reply:
So you suggest implementing this in pure python? So, where put the documentation that is required to create a QApplication object in the python main thread?

Reply:
I don't know.  I don't have strong feelings about it, especially with this sitting in a `qt` directory.